cargo-concordium/src/main.rs

```diff
@@ -888,35 +922,94 @@ fn handle_run_v1(run_cmd: RunCommand, module: &[u8]) -> anyhow::Result<()> {
                         };
 
                         match func_schema_opt {
+                            // V2 schemas don't have schemas for events.
                             Some(func_schema) => (
                                 true,
                                 func_schema.parameter(),
                                 func_schema.return_value(),
                                 func_schema.error(),
+                                None,
                             ),
-                            None => (true, None, None, None),
+                            None => (true, None, None, None, None),
                         }
                     }
-                    None => (true, None, None, None),
+                    None => (true, None, None, None, None),
+                }
+            }
+            Some(VersionedModuleSchema::V3(module_schema)) => {
+                // Getting the event schema from the init function if attached.
+                let schema_event = match module_schema.contracts.get(contract_name) {
+                    Some(contract_schema) => match contract_schema.init.as_ref() {
+                        Some(func_schema) => func_schema.event(),
+                        None => None,
+                    },
+                    None => None,
+                };
```

Comment:
The way the schemas are modeled now we do not seem to use the `event` schema in receive functions. That does not seem like the best solution.

If you changed the modelling of COntractV3 like I suggest in the https://github.com/Concordium/concordium-contracts-common/pull/54 comments then this will become a bit more direct.